src/transformers/generation_utils.py

```diff
@@ -863,6 +871,10 @@ def generate(
         model_kwargs["output_attentions"] = output_attentions
         model_kwargs["output_hidden_states"] = output_hidden_states
 
+        if encoder_outputs is not None:
```

Comment:
I would really like to keep if statements to an absolute minimum in generate, so I'm not a big fan of this to be honest

Reply:
If that's a concern, I can hide this logic into `_prepare_input_ids_for_generation` below. This would also make `_prepare_dummy_input_ids` unnecessary, as it's absorbed into `_prepare_input_ids_for_generation`.